lib/ui/painting/path_measure.cc

```diff
@@ -0,0 +1,121 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/lib/ui/painting/path_measure.h"
+
+#include <math.h>
+
+#include "flutter/lib/ui/painting/matrix.h"
+#include "lib/tonic/converter/dart_converter.h"
+#include "lib/tonic/dart_args.h"
+#include "lib/tonic/dart_binding_macros.h"
+#include "lib/tonic/dart_library_natives.h"
+
+using tonic::ToDart;
+
+namespace blink {
+
+typedef CanvasPathMeasure PathMeasure;
+
+static void PathMeasure_constructor(Dart_NativeArguments args) {
+  DartCallConstructor(&CanvasPathMeasure::Create, args);
+}
+
+IMPLEMENT_WRAPPERTYPEINFO(ui, PathMeasure);
+
+#define FOR_EACH_BINDING(V)          \
+  V(PathMeasure, setPath)            \
+  V(PathMeasure, getLength)          \
+  V(PathMeasure, getMatrix)          \
+  V(PathMeasure, getPosTan)          \
+  V(PathMeasure, getSegment)         \
+  V(PathMeasure, isClosed)           \
+  V(PathMeasure, nextContour)
+
+FOR_EACH_BINDING(DART_NATIVE_CALLBACK)
+
+void CanvasPathMeasure::RegisterNatives(tonic::DartLibraryNatives* natives) {
+  natives->Register({{"PathMeasure_constructor", PathMeasure_constructor, 3, true},
+                     FOR_EACH_BINDING(DART_REGISTER_NATIVE)});
+}
+
+fxl::RefPtr<CanvasPathMeasure> CanvasPathMeasure::Create(const CanvasPath* path, 
+                                                          bool forceClosed) {
+  fxl::RefPtr<CanvasPathMeasure> pathMeasure = fxl::MakeRefCounted<CanvasPathMeasure>();
+  if (path) {
+    const SkPath skPath = path->path();
+    pathMeasure->path_measure_ = new SkPathMeasure(skPath, forceClosed, 1);
+  } else {
+    pathMeasure->path_measure_ = new SkPathMeasure();
+  }
+  return pathMeasure;
+}
+
+CanvasPathMeasure::CanvasPathMeasure() {}
+
+CanvasPathMeasure::~CanvasPathMeasure() {
+  delete path_measure_;
+}
+
+void CanvasPathMeasure::setPath(const CanvasPath* path, 
+                                bool isClosed) {
+  const SkPath* skPath = &(path->path());
+  path_measure_->setPath(skPath, isClosed);
+}
+
+float CanvasPathMeasure::getLength() {
+  return path_measure_->getLength();
+}
+
+tonic::Float64List CanvasPathMeasure::getMatrix(float distance, 
+                                                int flags) {
+  SkMatrix matrix;
+  matrix.reset();
+  bool success = path_measure_->getMatrix(distance, &matrix, (SkPathMeasure::MatrixFlags)flags);
```

Comment:
Lets follow the [style guide](https://google.github.io/styleguide/cppguide.html#Casting) when casting (here and elsewhere).

Reply:
How does this method indicate failure? As written, failure returns the Identity matrix which may not be what the caller is expecting. Maybe the signature needs to be updated to account for a success flag?